projects/components/src/stepper/stepper.component.ts

```diff
@@ -138,14 +151,40 @@ export class StepperComponent implements AfterContentInit {
     }
   }
 
+  public nextOrSubmit(stepper: MatStepper): void {
+    const isLastStep = this.isLastStep(stepper);
+    if (isLastStep) {
+      this.submitted.emit();
+    } else {
+      stepper.next();
+    }
+  }
+
   public getActionButtonLabel(selectedIndex: number, steps: StepperTabComponent[]): string {
-    return steps[selectedIndex]?.actionButtonLabel ?? (selectedIndex === steps.length ? 'Submit' : 'Next');
+    const isLastStep = selectedIndex === steps.length - 1;
```

Comment:
nit - could work with stepper instead and reuse isLastStep. Doesn't really even need args.